### DISCUSSION TITLE
Does otPlatAlarmMilliStartAt() stack multiple timers?

### DISCUSSION THREAD

**Discussion by EricSTMicro** [openthread/openthread#8865] [Q&A] [answered]
Hello,

When OpenThread call the function `void otPlatAlarmMilliStartAt(otInstance *aInstance, uint32_t aT0, uint32_t aDt)`, I would like to know if there is an interest in keeping the former timers that may have been started by OpenThread but that have not been fired yet?

In other word, do we only have to care about one timer at a time at platform level? (I'm not speaking about the same function that is in microseconds). 

Regards,
Eric

**Comment by jwhui** [comment, marked as answer]
The `otPlatAlarmMilli` APIs are for a single alarm instance. OpenThread multiplexes multiple timers on top of that single instance - see [src/core/common/timer.hpp](https://github.com/openthread/openthread/blob/main/src/core/common/timer.hpp).

**Reply by abtink** [reply]
Hi @EricSTMicro 
- `otPlatAlram` is a a single timer.
- Subsequent call to `otPlatAlarmMilliStartAt()` replaces any previously scheduled timer values.

**Reply by EricSTMicro** [reply]
Ok thank you to both of you!
